src/HDF5.jl

```diff
@@ -482,9 +475,22 @@ end
 struct EmptyArray{T} end
 
 # Stub types to encode fixed-size arrays for H5T_ARRAY
-struct FixedArray{T,D} end
-size(::Type{FixedArray{T,D}}) where {T,D} = D
-eltype(::Type{FixedArray{T,D}}) where {T,D} = T
+struct FixedArray{T,D,L}
+  data::NTuple{L, T}
```

Comment:
how / where's this field being used ?

Reply:
This is used so that the julia type you construct for an `H5T_ARRAY` is memory compatible and can be directly written into.